### PR TITLE
Add DATE to dateadd macro

### LINKED ISSUE #405: Duck Db support of DATEADD from Core
I tried the example here: 
https://docs.getdbt.com/sql-reference/dateadd
```
SELECT {{ dateadd(datepart="month", interval=1, from_date_or_timestamp="'2021-08-12'") }}
```
This fails. 

I tweaked and this works: 
```
SELECT {{ dateadd(datepart="month", interval=1, from_date_or_timestamp="DATE '2021-08-12'") }}  AS period_of_load
```

In general I think adapters should support the documented case from core. 

Here duckdb seems to want date strings to be explicitly refererred to as dates in addition operations as per: https://duckdb.org/docs/sql/functions/date.html

As such I suggest that : 
https://github.com/duckdb/dbt-duckdb/blob/master/dbt/include/duckdb/macros/utils/dateadd.sql

Is updated to include: DATE at start

### PR DESCRIPTION
**Pull request by alittlesliceoftom**
Fixes #405 

tl;dr, helps you avoid errors like: 

```
  Binder Error: Could not choose a best candidate function for the function call "+(STRING_LITERAL, INTERVAL)". In order to select one, please add explicit type casts.        Candidate functions:
        +(DATE, INTERVAL) -> TIMESTAMP
        +(TIME, INTERVAL) -> TIME
        +(TIMESTAMP, INTERVAL) -> TIMESTAMP
        +(TIME WITH TIME ZONE, INTERVAL) -> TIME WITH TIME ZONE
        +(TIMESTAMP WITH TIME ZONE, INTERVAL) -> TIMESTAMP WITH TIME ZONE
        +(INTERVAL, INTERVAL) -> INTERVAL

  LINE 1: /* {"app": "dbt", "dbt_version": "1.7.9", "profile_...

```

### CODE REVIEW

**Comment by alittlesliceoftom** [comment]
See docs here: https://duckdb.org/docs/sql/functions/date.html

**Comment by alittlesliceoftom** [comment]
I didn't have time to setup integration tests, but tested the updated macro locally, and 'correct' example now works: 


 SELECT 
    {{ dateadd(datepart="month", interval=1, from_date_or_timestamp="'2021-08-12'") }}  AS period_of_load

**Comment by jwills** [comment]
yeah I'm surprised this didn't break on the regular unit tests-- like, does this not work anymore and I missed it? https://github.com/dbt-labs/dbt-adapters/blob/main/dbt-tests-adapter/dbt/tests/adapter/utils/fixture_dateadd.py

**Comment by alittlesliceoftom** [comment]
Possibly those are more explicit strings?

**Comment by jwills** [comment]
My concern here is what happens if I pass in a `TIMESTAMP` instead of a string or a `DATE`? Or a string that should be cast as a `TIMESTAMP` instead?

**Comment by alittlesliceoftom** [comment]
Yeh... Interestingly in DuckDb the dateadd operations seem to expect the coder to know and set the type ahead of time in the statement? 
https://duckdb.org/docs/sql/functions/interval

**Comment by jwills** [comment]
Closing this out as I believe the code was working the way dbt intended
